zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -7,51 +8,66 @@ use crate::{
 use eth_types::Field;
 use halo2_proofs::{
     circuit::{Layouter, SimpleFloorPlanner, Value},
-    plonk::{Circuit, ConstraintSystem, Error, Expression},
+    halo2curves::bn256::Fr,
+    plonk::{Advice, Circuit, Column, ConstraintSystem, Error, Fixed},
 };
-use mpt_zktrie::{hash::Hashable, operation::AccountOp, EthTrie, EthTrieCircuit, EthTrieConfig};
+use mpt_zktrie::mpt_circuits::{gadgets::poseidon::PoseidonLookup, mpt, types::Proof};
 
-/// re-wrapping for mpt circuit
-#[derive(Default, Clone, Debug)]
-pub struct MptCircuit<F: Field>(pub(crate) EthTrieCircuit<F, false>);
+impl PoseidonLookup for PoseidonTable {
+    fn lookup_columns_generic(&self) -> (Column<Fixed>, [Column<Advice>; 5]) {
+        (
+            self.q_enable,
+            [
+                self.hash_id,
+                self.input0,
+                self.input1,
+                self.control,
+                self.heading_mark,
+            ],
+        )
+    }
+}
+
+/// Circuit wrapped with mpt table data
+#[derive(Clone, Debug, Default)]
+pub struct MptCircuit<F: Field> {
+    row_limit: usize,
+    proofs: Vec<Proof>,
+    mpt_updates: witness::MptUpdates,
+    _phantom: std::marker::PhantomData<F>,
+}
 
 /// Circuit configuration argument ts
-pub struct MptCircuitConfigArgs<F: Field> {
+pub struct MptCircuitConfigArgs {
     /// PoseidonTable
     pub poseidon_table: PoseidonTable,
     /// MptTable
     pub mpt_table: MptTable,
     /// Challenges
-    pub challenges: Challenges<Expression<F>>,
+    pub challenges: Challenges,
 }
 
 /// re-wrapping for mpt config
-#[derive(Debug, Clone)]
-pub struct MptCircuitConfig(pub(crate) EthTrieConfig);
+#[derive(Clone)]
+pub struct MptCircuitConfig<F: Field>(
+    pub(crate) mpt::MptCircuitConfig,
+    pub(crate) MptTable,
+    std::marker::PhantomData<F>,
+);
 
-impl<F: Field> SubCircuitConfig<F> for MptCircuitConfig {
-    type ConfigArgs = MptCircuitConfigArgs<F>;
+impl SubCircuitConfig<Fr> for MptCircuitConfig<Fr> {
+    type ConfigArgs = MptCircuitConfigArgs;
 
     fn new(
-        meta: &mut ConstraintSystem<F>,
+        meta: &mut ConstraintSystem<Fr>,
         Self::ConfigArgs {
             poseidon_table,
             mpt_table,
             challenges,
         }: Self::ConfigArgs,
     ) -> Self {
-        let poseidon_table = (
-            poseidon_table.q_enable,
-            [
-                poseidon_table.hash_id,
-                poseidon_table.input0,
-                poseidon_table.input1,
-                poseidon_table.control,
-                poseidon_table.heading_mark,
-            ],
-        );
-        let mpt_table = (
-            mpt_table.q_enable,
+        let _mpt_table_inp = //(
```

Comment:
is this correct? @z2trillion

Reply:
yes, these constraints are currently still missing.

Reply:
https://github.com/scroll-tech/zkevm-circuits/pull/551  along with https://github.com/scroll-tech/mpt-circuit/pull/47 will fix this.